mep/common/views.py

```diff
@@ -0,0 +1,29 @@
+from django.views.generic.base import View, ContextMixin
+from django.core.paginator import Paginator
+
+class LabeledPagesMixin(ContextMixin):
+    '''View mixin to add labels for pages to a paginated view's context,
+    for use in rendering pagination controls.'''
+
+    def get_page_labels(self: View, paginator: Paginator):
+        '''Generate labels for pages. Defaults to labeling pages using numeric
+        ranges, e.g. `50-100`.'''
+        page_labels = []
+        # if there's nothing to paginate, just return an empty list
+        if paginator.count == 0:
+            return page_labels
+        for page in paginator.page_range:
+            page_start = (page - 1) * paginator.per_page
```

Comment:
Might be worth adding brief comments about the +1 / -1 stuff here to make it easier to read & understand when we come back to.